pandas/core/generic.py

```diff
@@ -3370,14 +3368,7 @@ class  max_speed
         new_data = self._data.take(
             indices, axis=self._get_block_manager_axis(axis), verify=True
         )
-        result = self._constructor(new_data).__finalize__(self)
-
-        # Maybe set copy if we didn't actually change the index.
```

Comment:
not sure if its relevant, but DTI.take sometimes returns a view

Reply:
In the case the indices can represent a slice, right? 
Personally I don't think we should make that optimization, as IMO it is much clearer for take to always be a copy, but I suppose for Index which is supposed to be immutable it might matter less.